federation/registry/cluster/etcd/etcd.go

```diff
@@ -39,8 +39,8 @@ func (r *StatusREST) New() runtime.Object {
 }
 
 // Update alters the status subset of an object.
-func (r *StatusREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
-	return r.store.Update(ctx, name, objInfo)
+func (r *StatusREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc) (runtime.Object, bool, error) {
```

Comment:
Instead of adding a billion parameters to a billion functions, how about making a struct that contains name, objInfo, and both of these validation functions? Then the next time we add/change something here we won't have to change every concrete implementation.

Reply:
> Instead of adding a billion parameters to a billion functions, how about making a struct that contains name, objInfo, and both of these validation functions? Then the next time we add/change something here we won't have to change every concrete implementation.

In this case, the compile failures are instrumental in chasing down the affected components.  We really don't want to miss one.  I think this is demonstrating a need to expand the generic etcd store in a way to support a standard status and scale.

Reply:
I would keep those changes out of this PR if we want to do them. We change a lot already, so let's try to leave those refactorings to a follow-up.